www/class/centreon-clapi/centreonHostTemplate.class.php

```diff
@@ -68,4 +68,36 @@ public function setinstance($parameters = null)
     {
         throw new CentreonClapiException(self::UNKNOWN_METHOD);
     }
+
+    /**
+     * Del Action
+     * Host template can be deleted only if it is not used in host
+     *
+     * @param string $objectName
+     * @return void
+     * @throws CentreonClapiException
+     */
+    public function del($objectName)
+    {
+        $sql = "SELECT COUNT(*) AS cnt
+                FROM host_template_relation htr 
+                JOIN host h ON h.host_id = htr.host_tpl_id 
+                WHERE h.host_name = :hostname";
+        $res = $this->db->query(
+            $sql,
+            array(
+                ':hostname' => $objectName
+            )
+        );
+        $result = (int)$res->fetch()['cnt'];
+        if ($result > 0) {
+            throw new CentreonClapiException(self::OBJECTALREADYLINKED . ":" . $objectName);
+        } else {
+            parent::del($objectName);
```

Comment:
If there is more than one change in the database, please to create a transaction to include these two deletions.

Reply:
This happens across entire project.